qiskit_experiments/database_service/device_component.py

```diff
@@ -35,6 +38,9 @@ def __init__(self, index: int):
     def __str__(self):
         return f"Q{self._index}"
 
+    def __json_encode__(self):
```

Comment:
`__json_decode__` is not necessary?

Reply:
It's not necessary for these classes since they can be reconstructed from the default method of init kwargs.